models/user.go

```diff
@@ -20,6 +21,22 @@ const (
 	usersTable string = "users"
 )
 
+//Authenticates the user's credentials and returns the user
+func (db *DB) CheckUser(user *User) (*User, error) {
+	userFromDB := &User{}
+	queryString := fmt.Sprintf("SELECT ID, Name, Password FROM %s WHERE EMAIL=?", usersTable)
+	if err := db.QueryRow(queryString, user.Email).Scan(&(userFromDB.ID),
+		&(userFromDB.Name), &(userFromDB.Password)); err != nil {
+		return nil, err
+	}
+	err := checkPasswordHash(user.Password, userFromDB.Password)
+	if err != nil {
+		err := errors.New("password incorrect")
+		return nil, err
```

Comment:
Just return the new error straight-up to make it cleaner.
```suggestion
		return nil, errors.New("password incorrect")
```